File: blind-run-backend/src/main/java/com/example/demo/entity/BlindProfile.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 盲人用户资料实体 —— 对应数据库中的 blind_profile 表
 */
@Data
@Entity
@Table(name = "blind_profile")
public class BlindProfile {

    @Id
    private Long userId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "user_id")
    private User user;

    private String name;

    @Column(name = "emergency_contact_name")
    private String emergencyContactName;

    @Column(name = "emergency_contact_phone")
    private String emergencyContactPhone;

    @Column(name = "emergency_contact_relation")
    private String emergencyContactRelation;

    @Column(name = "running_pace")
    private String runningPace;

    @Column(name = "special_needs")
    private String specialNeeds;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: blind-run-frontend/lib/core/models/blind_profile.dart
class BlindProfile {
  const BlindProfile({
    this.name = '',
    this.runningPace = '',
    this.specialNeeds = '',
  });

  final String name;
  final String runningPace;
  final String specialNeeds;

  BlindProfile copyWith({
    String? name,
    String? runningPace,
    String? specialNeeds,
  }) {
    return BlindProfile(
      name: name ?? this.name,
      runningPace: runningPace ?? this.runningPace,
      specialNeeds: specialNeeds ?? this.specialNeeds,
    );
  }
}
